crates/ra_hir_def/src/body/lower.rs

```diff
@@ -575,9 +575,16 @@ impl ExprCollector<'_> {
             self.body.item_scope.define_def(def);
             if let Some(name) = name {
                 let vis = crate::visibility::Visibility::Public; // FIXME determine correctly
-                self.body
-                    .item_scope
-                    .push_res(name.as_name(), crate::per_ns::PerNs::from_def(def, vis));
+                let favor_types = match def {
```

Comment:
I think it'd be clearer to invert this and call it `has_constructor`. maybe even move this logic into `from_def`? It's not so much "favoring" types as whether the def exists in the value namespace at all.

Reply:
Originally this logic is inside `from_def`, however, in collecting phase, I can't do the same as it will break the `no db change rules`.

+1 for `has_constructor`

Reply:
Fixed for `has_constructor`